test/snippet/stream/transparent_istream.cpp

```diff
@@ -0,0 +1,41 @@
+#include <fstream>
+
+#include <bio/stream/transparent_istream.hpp>
+#include <bio/stream/transparent_ostream.hpp>
+
+int main()
+{
+    { // create my_file.txt so the code below works
+        std::ofstream os{"my_file.txt"};
+        os << "Foo";
+    }
+
+    { // create my_file.txt.gz so the code below works
+        bio::transparent_ostream os{"my_file.txt.gz"};
+        os << "Foo";
+    }
+
+    {
+    //![construction]
+    std::string buffer;
+
+    bio::transparent_istream s1{"my_file.txt"};      // behaves like std::ifstream
+    s1 >> buffer;
+
+    bio::transparent_istream s2{"my_file.txt.gz"};   // file is transparently decompressed
+    s2 >> buffer;
+
+    // bio::transparent_istream s3{std::cin};        // wrap standard input
+    // s3 >> buffer;
+    //![construction]
```

Comment:
Note: this would make the snippet tests to wait endlessly for command line input. Don't now how you want to do this.

Reply:
You can switch the readbuffer of std::cin with that of an ifstream inside the test:
```cpp
    std::ifstream in{"data/input.txt"};
    std::cin.rdbuf(in.rdbuf());
```